source/codegen/templates/service_helpers.mako

```diff
@@ -777,7 +778,11 @@ ${set_response_values(normal_outputs, init_method)}\
       %for error_parameter in error_parameters:
       context->AddTrailingMetadata("${error_parameter["return_on_error_key"]}", std::to_string(${common_helpers.pascal_to_snake(error_parameter["name"])}));
       %endfor
+      %if len(error_parameters)>0 and is_read_function:
+      if (!status_ok(status) && status != DAQmxErrorSamplesNotYetAvailable) {
```

Comment:
Instead of a DAQmx-specific hack, I think we should add a key to the function metadata that opts in to this feature and specifies the timeout error constant. Other drivers such as [nisync](https://github.com/ni/grpc-device/blob/main/source/custom/nisync_service.custom.cpp) could benefit.

How about `'timeout_error': 'DAQmxErrorSamplesNotYetAvailable'`?

Reply:
Looks good